### ISSUE TITLE
FileWrite error when running multiple simultanius checks

### ISSUE THREAD

**Issue by larsbloch** [Battle-Line-Productions/EmailValidator#11]
Hello again

I get erros when running multiple checks on emails.

    private async Task DownloadDisposableEmailFile()
    {
        var client = _httpClientFactory.CreateClient(DisposableEmailClientName);

        HttpResponseMessage result;

        try
        {
            result = await client.GetAsync(DisposableEmailDomainUri);
        }
        catch (Exception ex)
        {
            throw new Exception("Failed to download disposable email file", ex);
        }

        if (result.IsSuccessStatusCode)
        {
            var content = await result.Content.ReadAsStringAsync();
            var lines = content.Split(new[] { "\r\n", "\r", "\n" }, StringSplitOptions.None);

            try
            {
                await File.WriteAllLinesAsync(DisposableEmailFile, lines);
            }
            catch (Exception ex)
            {
                throw new Exception("Failed to write disposable email file to disk", ex);
            }
        }
    }
    
    
    This error
    throw new Exception("Failed to write disposable email file to disk", ex);
    
    It seems like it wants to write the contents to disk after each check
    
    Also it does not seem to work to disable the disposable email check

**Comment by funkel1989** [comment]
Can you confirm it downloaded the file to your file system?

Check dependencies checks to see if the file exists first before downloading the file or if the file is greater than a day old.

it should not try to download again unless either of those criteria were met. I was able to run multiple emails through using singleton, transient and scoped services in a look in the same execution and via multiple executions.

I'm going to need more code from you to understand what your doing that could be causing this? 

There is a scenario I don't have time to test right now regarding multi-threading and potentially disposable being kicked off more than once so I'm pushing a change to prevent this just incase that might be your issue.

also I've moved the dependencies download to only occur when disposable is on. new version will come once my pipeline is done and i can issue the release. maybe an hour?

**Comment by larsbloch** [comment]
I will test with your new version and report back with more code if needed. It seems your solution could be the solution.

Btw something completely different. I have another library which does a smtp check agains the email address. Is it something that you have thought about adding, or is within scope of your goal with the library ? 
I could do the research and provide the code if you would implement it ?

**Comment by funkel1989** [comment]
I was going to implement it but determined it wasn't worth while. The majority of correctly configured SMTP relays will fail that check or return a response that is unhelpful. I actually got a text response from gmail with a javascript library that does that and the test response said "why not just try to send an email?"

After much research I've learned the only true way to validate an email address is to send it and get a response from the user, everything else is just mitigating efforts.
